Lifespark-Technologies/Infomed
# Allow proxying to the backend server

This change allows to proxy any HTTP request unknown to the backend server on port 8000.

To test this change:

1. Run the backend server on port 8000.
2. Run the frontend development server (`npm install`, `npm start`).
3. Go to http://localhost:3000, if you haven't been taken there automatically. Note that this is the app version from the frontend server.
4. Go to Chrome Developer Console (or an analogous tool).
5. Make any request to the backend, but through port 3000 (example: `fetch('http://localhost:3000/api/lead').then(req => req.json()).then(j => console.log(j));`). Observe results on the console; you should see the response from the API. Note that simply pasting URL into the browser won't yield the same results; the proxy works only if the request doesn't contain a `text/html` header in the request. Fetching the URL using `curl` should work out of the box.

For details, see https://create-react-app.dev/docs/proxying-api-requests-in-development/.
